### ISSUE TITLE
onnxruntime latest version segment fault

### ISSUE THREAD

**Issue by henrywu2019** [microsoft/onnxruntime#10113]
**Describe the bug**
in docker, when cpu-sets are specified, onnxruntime will get segment fault.

**Urgency**
affect production

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Centos 7
- ONNX Runtime installed from (source or binary): pypi
- ONNX Runtime version: 1.10.0
- Python version: 3.8
- CUDA/cuDNN version: CPU
- GPU model and memory: CPU intel/AMD

** Error **

```
RuntimeError: /onnxruntime_src/onnxruntime/core/platform/posix/env.cc:183 
onnxruntime::{anonymous}::PosixThread::PosixThread(const char*, int, 
unsigned int (*)(int, Eigen::ThreadPoolInterface*), Eigen::ThreadPoolInterface*,
const onnxruntime::ThreadOptions&) pthread_setaffinity_np failed, error code: 0 error msg:
```

**Comment by henrywu2019** [comment]
btw, version 1.6.0 works well

**Comment by chausner** [comment]
I wonder if this could also somehow be related to https://github.com/microsoft/onnxruntime/issues/10104.

**Comment by snnn** [comment]
Right. It doesn't work in that way. Do you know if Linux has an API to get known if the available CPUs were restricted to a subset? 
And I suggest we add an SessionOption for thread affinity, and turn if off by default.

**Comment by chausner** [comment]
> Do you know if Linux has an API to get known if the available CPUs were restricted to a subset?

There's [pthread_getaffinity_np](https://www.man7.org/linux/man-pages/man3/pthread_setaffinity_np.3.html) and the pseudo-filesystem [/dev/cpuset](https://www.man7.org/linux/man-pages/man7/cpuset.7.html).

**Comment by chausner-audeering** [comment]
@henrywu2019 I think onnxruntime currently does not check for the correct size of the affinity vector. It assumes to have at least as many entries as threads in the thread pool. I have proposed an additional check in https://github.com/microsoft/onnxruntime/pull/10122. Could this maybe be the reason for the crash you are experiencing?

**Comment by snnn** [comment]
When you use cpuset with onnxruntime, you can't use onnxruntime's default session options. You need to explicitly tell onnxruntime how many threads it should create.

**Comment by henrywu2019** [comment]
Can you give me some workable sample code? Thanks! @snnn

**Comment by snnn** [comment]
Please set this property: https://onnxruntime.ai/docs/api/python/api_summary.html#onnxruntime.SessionOptions.intra_op_num_threads

**Comment by henrywu2019** [comment]
set it equal to the number of the really available cores?

**Comment by snnn** [comment]
Yes.

**Comment by henrywu2019** [comment]
Do I need to specify which cpu cores?

**Comment by serser** [comment]
It turns out it simply needs more cpu & memory, this problem disappears when I tremendously increased them.
